### PR TITLE
pipeline datatypes

### PR DESCRIPTION
**Pull request by emlynjdavies**
making draft PR to enable RTD builds

### CODE REVIEW

File: pyopia/instrument/silcam.py [hunk 1]
```diff
@@ -23,41 +25,63 @@ def timestamp_from_filename(filename):
...
+    Returns
+    -------
+        data.imc : :class:`ImagePrepData`
+    '''
 
     def __init__(self):
         pass
 
-    def __call__(self, data):
+    def __call__(self, data: ImagePrepData) -> ImagePrepData:
```
Comment (Asteinvik) [inline pyopia/instrument/silcam.py:82]: I think this should take LoadData as input and only return ImagePrepData. The difference between them is imc, and that is created here

File: pyopia/instrument/silcam.py [hunk 2]
```diff
@@ -23,41 +25,63 @@ def timestamp_from_filename(filename):
...
         # @todo
-        # #imbg = data['imbg']
+        # #imbg = data.imbg
         # background correction
         print('WARNING: Background correction not implemented!')
         imraw = data['img']
```
Comment (Asteinvik) [inline pyopia/instrument/silcam.py:87]: use data.img here

File: pyopia/instrument/silcam.py [hunk 3]
```diff
@@ -23,41 +25,63 @@ def timestamp_from_filename(filename):
...
     '''
 
     def __init__(self, filename):
         self.filename = filename
 
-    def __call__(self, data):
+    def __call__(self, data: LoadData) -> LoadData:
         timestamp = timestamp_from_filename(self.filename)
         img = np.load(self.filename, allow_pickle=False)
         data['timestamp'] = timestamp
```
Comment (Asteinvik) [inline pyopia/instrument/silcam.py:56]: Here you can create the LoadData by doing `data = LoadData(timestamp=timestamp, img=img)`

File: pyopia/instrument/silcam.py [hunk 4]
```diff
@@ -23,41 +25,63 @@ def timestamp_from_filename(filename):
...
-        raw silcam image
+        data.img : :class:`LoadData`
+        data.timestamp : :class:`LoadData`
     '''
 
     def __init__(self, filename):
         self.filename = filename
 
-    def __call__(self, data):
+    def __call__(self, data: LoadData) -> LoadData:
```
Comment (Asteinvik) [inline pyopia/instrument/silcam.py:53]: This does not use anything from data, so to show this you can do 
`def __call__(self, _ ) -> LoadData:`
The underscore argument implies that this input is not used

File: pyopia/instrument/silcam.py (hunk 3 above)
Comment (emlynjdavies) [reply pyopia/instrument/silcam.py:56]: doing this results in the output of this removing the other data that is not used in this function (in this case the classifier object)

**Pull request closed without merging**